pandas/core/arrays/interval.py

```diff
@@ -284,7 +306,10 @@ def _simple_new(
         from pandas.core.indexes.base import ensure_index
 
         left = ensure_index(left, copy=copy)
+        left = maybe_convert_numeric_to_64bit(left)
```

Comment:
just for my understanding, what's an example of where this makes a difference? the test you've modifed passes even without this change

Reply:
I think I took the wrong approach here originally.

The issue that this was supposed to solve is that on 32-bit systems  e.g. `IntervalArray.from_breaks([1, 2, 3])` should give an array with dtype `interval[int64, right]` to align with the convention that lists in constructors should interpreted as 64-bit (e.g. ``Series([1, 2, 3])` and `Index([1, 2, 3])` both give int64 dtype even on 32-bit systems). Previously, (after #49560) giving lists to `IntervalArray` gave `interval[int32, right]`. This affected some tests in #49560 which is the reason I hava taken this up.

In the newest version I moved this logic to a `_maybe_convert_platform_interval`, which IMO should be the better location fir this.

Reply:
Also, this issue on 32-bit systems is only with integer dtypes as `np.asarray([1.5])` will always have float64 dtype. So I've made this simpler in the newest version by just checking for integer dtype and converting to `int64` if needed.